### PR TITLE
feat: handle transparency

### PR DESCRIPTION
**Pull request by borisdayma**
Converts transparent pixels to white (currently they become black).

Raw image with transparent pixels
logo (1)

Current behavior
test_logo

New behavior
image

### CODE REVIEW

File: img2dataset/downloader.py
Comment (Skylion007) [inline img2dataset/downloader.py:83]: This is going to leave some rough matting on certain PNGs. I don't thine cv2 color convert does alpha matting, does it? The issue with this is that it will work on certain PNGs, but some PNGS are not binary alpha masks and for this it will not work well. You should do a proper alpha matting with OpenCV to ensure that it handles semitransparent matting as well.

**Review by Skylion007 — CHANGES_REQUESTED** [review]

File: img2dataset/downloader.py
Comment (borisdayma) [reply img2dataset/downloader.py:83]: Right now I'm just changing transparency from black to white. It's currently being converted to black.
I didn't understand which issue you were talking about, feel free to propose any improvement.
Do you have a sample image where this change is an issue?

Comment (borisdayma) [reply img2dataset/downloader.py:83]: I tried on the image from https://github.com/np-csu/AlphaMatting

With or without this PR I'm getting this:

Comment (Skylion007) [reply img2dataset/downloader.py:83]: You wouldn't detect non-binary binary masks, you would just do proper alpha matting by multiplying the RGB by the alpha channel and multiplying the bg color by (1 - alpha) and summing them together. (assumes normalized 0-1 alpha).

**Comment by rom1504** [comment]
can you fix the lint ? (run make black then make lint)

**Comment by borisdayma** [comment]
Confused with the error.
When I run the command that fails in the build, I don't get an error.

```bash
~/img2dataset$ python -m black --check -l 120 img2dataset
All done! ✨ 🍰 ✨
3 files would be left unchanged.
```

**Comment by rom1504** [comment]
yeah you have the wrong version of black, run `pip install -r requirements-test.txt` to install the right one

**Comment by borisdayma** [comment]
Hold off a bit before merging, I'm gonna try to apply the suggestions from @Skylion007

File: img2dataset/downloader.py
Comment (borisdayma) [reply img2dataset/downloader.py]: Thanks, I applied your suggestions

**Comment by borisdayma** [comment]
Should be good now!

I tested on these:
* https://i.imgur.com/B922OoM.png
* https://raw.githubusercontent.com/borisdayma/dalle-mini/main/img/logo.png
* some regular images

File: img2dataset/downloader.py
Comment (Skylion007) [inline img2dataset/downloader.py:85]:
```suggestion
                img = np.rint(img.clip(min=0, max=255)).astype(np.uint8)
```
Add rounding

**Review by Skylion007 — APPROVED** [review]
Approval with nit

File: img2dataset/downloader.py
Comment (Skylion007) [inline img2dataset/downloader.py:81]: You may to want to make the alpha color configurable. Otherwise, this looks good. I also added a comment about adding proper rounding.

Comment (borisdayma) [reply img2dataset/downloader.py:81]: What do you think @rom1504 ?
You want to be able to choose the background color (instead of white) or we wait to see if it's requested?

Comment (rom1504) [reply img2dataset/downloader.py:81]: no I think let's keep white only for now so it's coherent with the resize border color

we may add that option if anybody can think of any use, let's not add it now

**Comment by rom1504** [comment]
should be ok but can you check if the speed is the same before/after ?
(I intend to add that to the CI after #65 but for now we still need to check manually)

**Comment by borisdayma** [comment]
There's probably not that very images with 4 channels to test the effect.

Anyway I tested on 500k images but my connection may be unstable:
* before this PR: 674 img/sec
* with this PR: 2 runs - 620 img/sec and 705 img/sec (same command, same data)

**Comment by rom1504** [comment]
Thanks!

**Pull request merged**
